File: app/src/main/java/com/example/andrasnemeth/nnotification/quicksettings/NNotificationTileService.java
package com.example.andrasnemeth.nnotification.quicksettings;

import android.service.quicksettings.Tile;
import android.service.quicksettings.TileService;
import android.widget.Toast;

import com.example.andrasnemeth.nnotification.R;

/**
 * Created by guni8 on 2016. 04. 05..
 */
public class NNotificationTileService extends TileService {

    @Override
    public void onTileAdded() {
        super.onTileAdded();
    }

    @Override
    public void onStartListening() {
        super.onStartListening();
    }

    @Override
    public void onClick() {
        super.onClick();
        Tile tile = getQsTile();
        if (tile.getState() == Tile.STATE_ACTIVE) {
            tile.setState(Tile.STATE_INACTIVE);
            tile.setLabel(getString(R.string.notification_off));
            Toast.makeText(NNotificationTileService.this, "Tile turned off", Toast.LENGTH_SHORT).show();
        } else {
            tile.setState(Tile.STATE_ACTIVE);
            tile.setLabel(getString(R.string.notification_on));
            Toast.makeText(NNotificationTileService.this, "Tile turned on", Toast.LENGTH_SHORT).show();
        }
        tile.updateTile();
    }
}

File: app/src/main/java/com/example/andrasnemeth/nnotification/quicksettings/NNotificationTileService.kt
package com.example.andrasnemeth.nnotification.quicksettings

import android.service.quicksettings.Tile
import android.service.quicksettings.TileService
import android.widget.Toast
import com.example.andrasnemeth.nnotification.R

/**
 * Created by guni8 on 2016. 04. 05..
 */
class NNotificationTileService : TileService() {
    override fun onTileAdded() {
        super.onTileAdded()
    }

    override fun onStartListening() {
        super.onStartListening()
    }

    override fun onClick() {
        super.onClick()
        val tile = qsTile
        if (tile.state == Tile.STATE_ACTIVE) {
            tile.state = Tile.STATE_INACTIVE
            tile.label = getString(R.string.notification_off)
            Toast.makeText(this@NNotificationTileService, "Tile turned off", Toast.LENGTH_SHORT).show()
        } else {
            tile.state = Tile.STATE_ACTIVE
            tile.label = getString(R.string.notification_on)
            Toast.makeText(this@NNotificationTileService, "Tile turned on", Toast.LENGTH_SHORT).show()
        }
        tile.updateTile()
    }
}